contribution/translation/prevalidator.php

```diff
@@ -67,282 +64,98 @@ public function get_helper()
 	}
 
 	/**
-	 * Checks the file for the array contents
-	 * Make sure it has all the keys present in the newest version
+	 * Run the phpBB Translation Validator
 	 *
 	 * @param \phpbb\titania\entity\package $package
-	 * @param string $reference_filepath The path to the files against I want to validate the uploaded package
-	 * @return array Returns an array of error messages encountered
+	 * @param string $origin_iso
+	 * @param string $phpbb_version
+	 * @return int Returns an array of error messages encountered
 	 */
-	public function check_package($package, $reference_filepath)
+	public function check_package($package, $origin_iso, $phpbb_version)
 	{
-		$package->ensure_extracted();
-		$error = $missing_keys = array();
-
-		// Basically the individual parts of the translation, we check them separately, because they have colliding filenames
-		$types = array(
-			'language' => 'language/',
-			'prosilver' => 'styles/prosilver/imageset/en',
-			'subsilver2' => 'styles/subsilver2/imageset/en',
-		);
+		$results = '';
```

Comment:
Apparently, this is not needed here. It gets assigned below no matter what.

Also we should be using the language class now instead of user class for lang functions.

Reply:
Done @ 8fb5aec912a6dacb7db05a20dd590ae5253f2cc1